pkg/prometheus/statefulset.go

```diff
@@ -779,14 +776,18 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 			disableCompaction = true
 		}
 
-		if p.Spec.Thanos.TracingConfig != nil {
-			container.Args = append(container.Args, "--tracing.config=$(TRACING_CONFIG)")
-			container.Env = append(container.Env, v1.EnvVar{
-				Name: "TRACING_CONFIG",
-				ValueFrom: &v1.EnvVarSource{
-					SecretKeyRef: p.Spec.Thanos.TracingConfig,
-				},
-			})
+		if p.Spec.Thanos.TracingConfig != nil || p.Spec.Thanos.TracingConfigFile != nil {
+			if p.Spec.Thanos.TracingConfigFile != nil {
+				container.Args = append(container.Args, "--tracing.config-file="+*p.Spec.Thanos.TracingConfigFile)
+			} else {
```

Comment:
see above regarding the if-else construct.

Reply:
Ok.